qiskit/providers/models/pulsedefaults.py

```diff
@@ -94,44 +98,327 @@ class Command(BaseModel):
+
+        Args:
+            operation: The operation for which to look.
+            qubits: The specific qubits for the operation.
+
+        Returns:
+            True iff the operation is defined.
+        """
+        return operation in self._ops_def and \
+            _to_tuple(qubits) in self._ops_def[operation]
+
+    def assert_has(self, operation: str, qubits: Union[int, Iterable[int]]) -> None:
+        """
+        Convenience method to check that the given operation is defined, and error if it is not.
 
-    def build_cmd_def(self):
-        """Construct the `CmdDef` object for the backend.
+        Args:
+            operation: The operation for which to look.
+            qubits: The specific qubits for the operation.
 
         Returns:
-            CmdDef: `CmdDef` instance generated from defaults
+            None
+
+        Raises:
+            PulseError: If the operation is not defined on the qubits.
+        """
+        if not self.has(operation, _to_tuple(qubits)):
+            if operation in self._ops_def:
+                raise PulseError("Operation '{op}' exists, but is only defined for qubits "
+                                 "{qubits}.".format(op=operation,
+                                                    qubits=self.qubits_with_op(operation)))
+            raise PulseError("Operation '{op}' is not defined for this "
+                             "system.".format(op=operation))
+
+    def get(self,
+            operation: str,
+            qubits: Union[int, Iterable[int]],
+            *params: List[Union[int, float, complex]],
+            **kwparams: Dict[str, Union[int, float, complex]]) -> Schedule:
+        """
+        Return the defined Schedule for the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+            *params: Command parameters for generating the output schedule.
+            **kwparams: Keyworded command parameters for generating the schedule.
+
+        Returns:
+            The Schedule defined for the input.
+        """
+        self.assert_has(operation, qubits)
+        schedule = self._ops_def[operation].get(_to_tuple(qubits))
+        if isinstance(schedule, ParameterizedSchedule):
+            schedule = schedule.bind_parameters(*params, **kwparams)
+        return schedule
+
+    def get_parameters(self, operation: str, qubits: Union[int, Iterable[int]]) -> Tuple[str]:
+        """
+        Return the list of parameters taken by the given operation on the given qubits.
+
+        Args:
+            operation: Name of the operation.
+            qubits: The qubits for the operation.
+
+        Returns:
+            The names of the parameters required by the operation.
+        """
+        self.assert_has(operation, qubits)
+        return self._ops_def[operation][_to_tuple(qubits)].parameters
+
+    def add(self,
+            operation: str,
```

Comment:
Should `operation` be `instruction`? In the future, we could have these accept the circuit `Instruction`? This would be similar for other functions in this class.

Reply:
Yes, I made a comment in the slack channel about how all of these had to change. This is why I haven't gotten to it yet

Reply:
Why would you want it to accept an `Instruction`? There is nothing we need here but the name

Reply:
But -- let's put this into another github issue, for tracking and for discussions